datumaro/components/dataset.py

```diff
@@ -956,9 +956,11 @@ def import_from(cls, path: str, format: Optional[str] = None, *,
         for src_conf in detected_sources:
             if not isinstance(src_conf, Source):
                 src_conf = Source(src_conf)
-            extractors.append(env.make_extractor(
+            extractor = env.make_extractor(
                 src_conf.format, src_conf.url, **src_conf.options
-            ))
+            )
+            cls.check_media_type(extractor)
```

Comment:
Probably, this call is better be moved to `init_cache()`.

Reply:
Moved to `init_cache()`

Reply:
The current variant also is not good because it does full dataset traversal, which can be a performance issue. Please try to integrate it into `_iter_init_cache`.

Reply:
Moved to `_iter_init_cache`